src/asset_manager/view/FileUploader.js

```diff
@@ -115,12 +115,20 @@ export default Backbone.View.extend(
         body.append(param, params[param]);
       }
 
+      var fileSize = 0;
+
       if (this.multiUpload) {
         for (let i = 0; i < files.length; i++) {
           body.append(`${config.uploadName}[]`, files[i]);
+          fileSize += files[i].size;
         }
       } else if (files.length) {
         body.append(config.uploadName, files[0]);
+        fileSize += files[0].size;
+      }
+      if (config.uploadMaxFileSize > 0 && config.uploadMaxFileSize <= fileSize / 1024) {
+        alert(config.uploadMaxFileSizeAlertText || 'The file(s) you have selected to add exceed your space limit');
+        return;
       }
```

Comment:
IMHO, instead of those 2 options, I'd prefer something like `config.beforeUpload({ fileSize, files, ...maybeOtherOptions })` callback where the user can define its own logic and raise its own UI (we shouldn't use `alert` in the framework, if you feel the need to do it, it means you have to create a customizable callback for the user). Eg.
```js
const { beforeUpload } = config;
const beforeUploadRes = beforeUpload && 
	beforeUpload({ fileSize, files, ...maybeOtherOptions });
if (beforeUploadRes === false) return;
```